driver-mapping/src/main/java/com/datastax/driver/mapping/MappingManager.java

```diff
@@ -74,6 +74,78 @@ public MappingManager(Session session, ProtocolVersion protocolVersion) {
         // which nodes might join the cluster later.
         // At least if protocol >=2 we know there won't be any 1.2 nodes ever.
         this.isCassandraV1 = (protocolVersion == ProtocolVersion.V1);
+        session.getCluster().register(new SchemaChangeListenerBase() {
+
+            @Override
+            public void onTableRemoved(TableMetadata table) {
+                synchronized (mappers) {
+                    Iterator<Mapper<?>> it = mappers.values().iterator();
+                    while (it.hasNext()) {
+                        Mapper<?> mapper = it.next();
+                        if (mapper.getTableMetadata().equals(table)) {
+                            LOGGER.error("Table {} has been removed; existing mappers for @Entity annotated {} will not work anymore", table.getName(), mapper.getMappedClass());
+                            it.remove();
+                        }
+                    }
+                }
+            }
+
+            @Override
+            public void onTableChanged(TableMetadata current, TableMetadata previous) {
```

Comment:
Ah maybe it is needed in the UDT case because it updates the Codec, where no analog is needed for table.  I suppose only benefit is maybe we'd be able to log the error here instead of when the user tries to use the mapper.